### PR TITLE
Use kgen to insert avro upsert data in benchmark

### PR DESCRIPTION
**Pull request by cirego**
Use `kgen` as the generator for `upsert-avro-test`. This dramatically reduces the amount of time required to generate the dataset for the upsert benchmark.

This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/5511)

### CODE REVIEW

File: src/kafka-util/src/bin/kgen.rs
Comment (benesch) [inline src/kafka-util/src/bin/kgen.rs:522]: Don't be afraid to just stick an `if` statement after the option parsing! There's only so much these magic clap configurations can do IME.

File: test/benchmark/avro_upsert/generator/generate_upserts.py
Comment (benesch) [inline test/benchmark/avro_upsert/generator/generate_upserts.py:89]: I'm surprised that wrapping the Rust in the Python was required for performance! Like, I'd expect the Rust to just be fast enough to saturate the network. Should be super easy to spawn a little thread pool with Rayon if required.

Anyway, 100% not worth changing around right this second, but food for thought for the future!

**Review by benesch — APPROVED** [review]

File: src/kafka-util/src/bin/kgen.rs
Comment (umanwizard) [inline src/kafka-util/src/bin/kgen.rs:521]: https://github.com/clap-rs/clap/discussions/2039

**Review by umanwizard — APPROVED** [review]

File: test/benchmark/avro_upsert/generator/generate_upserts.py
Comment (umanwizard) [reply test/benchmark/avro_upsert/generator/generate_upserts.py:89]: IME, generating all these random values and then populating the avro object becomes the bottleneck before a good network connection does.

Comment (cirego) [reply test/benchmark/avro_upsert/generator/generate_upserts.py:89]: I can look at using Rayon -- I was only using Python as a replacement for `xargs` or `parallel`, as well as reading the schema / distribution files send those into `kgen`.

I'll make a new PR to use `rayon` . What do we think of allowing the `schema` / `distribution` arguments to be filenames?

File: src/kafka-util/src/bin/kgen.rs
Comment (cirego) [reply src/kafka-util/src/bin/kgen.rs:522]: Sounds good, will do!

File: test/benchmark/avro_upsert/generator/generate_upserts.py
Comment (umanwizard) [reply test/benchmark/avro_upsert/generator/generate_upserts.py:89]: I think it's fine to script the parallelism in Python for now (also talked about this with Nikhil a few mins ago). If we ever start doing this more often though, we can add parallelism to the script itself using Rayon or similar.

Comment (cirego) [reply test/benchmark/avro_upsert/generator/generate_upserts.py:89]: Sounds good. I'll go ahead and merge this as is!

**Comment by benesch** [comment]
Whoops, sorry for the `num-messages`/`num-records` snafu. I didn't mean to change that. Arguably we should just fix that back to the way it was in kgen but totally your call.

**Comment by cirego** [comment]
> Whoops, sorry for the `num-messages`/`num-records` snafu. I didn't mean to change that. Arguably we should just fix that back to the way it was in kgen but totally your call.

Messages probably makes more sense in Kafka land, but I'm ambivalent. It was an easy enough fixup.

I did, however, file #5571 to make sure that we don't regress on these things again (once #5511 lands).

**Pull request merged**
